### PR TITLE
Add stock summary retrieval and decoding functionality

### PR DESCRIPTION
**Pull request by elminson**
**Description**
Adding stock Summary to get information about shares and stock data 
I need this information to get the float shares of a stock
If I need this information maybe another one needs  it

### CODE REVIEW

**Comment by scheb** [comment]
Hi, could you please explain why this is needed. Isn't `getQuote` doing the same thing?

**Comment by elminson** [comment]
Hi @scheb thank you for your quick response, 
`getQuote` don't have some important information (in my case is critical to get the float shares, operatingCashflow, sharesShortPriorMonth, freeCashflow, etc).

I have 2 options, 
1) Create a new method (`stockSummary`) 
2) Add what I need to `getQuote` method. 

Because was easier to tests and less disruptive to the current method,  I had pick option 1

Let me know what do you think about it.

Here are the data for `AAPL` called by each method 
stockSummary.json  => https://gist.github.com/elminson/e996c7bbcd21c066406b651bb061775f

getQuote.json => https://gist.github.com/elminson/0d1e2afaba3b815de5cc06cc1fc1813c

**Comment by scheb** [comment]
Okay, I see. Could you please do me a favor and rebase your changes onto the main branch and force-push an update? For some reason the CI didn't trigger. I hope it will after the push. I'm pretty sure there will be some code style issues popping up that we'd need to fix before we can merge.

**Comment by elminson** [comment]
@scheb I think now should be good if not let me know

**Comment by scheb** [comment]
Please check the coding standards job, there have been some issues popping up.

**Comment by elminson** [comment]
@scheb all code standards now are passing on my local please confirm :)

File: src/ApiClient.php [hunk 1]
```diff
@@ -249,4 +249,25 @@ private function getRandomQueryServer(): int
...
     }
+
+    public function stockSummary(string $symbol): array
+    {
+        $qs = $this->getRandomQueryServer();
+        $cookieJar = new CookieJar();
+
+        // Initialize session cookies
+        $initialUrl = 'https://fc.yahoo.com';
+        $modules = 'financialData,quoteType,defaultKeyStatistics,assetProfile,summaryDetail';
```
Comment (scheb) [inline src/ApiClient.php]: Readability suggestion: Please move this down to the "fetch quotes" where it's actually used.

**Review by scheb — COMMENTED** [review]
One more thing, then we're good to go.

File: src/ApiClient.php (hunk 1 above)
Comment (elminson) [reply src/ApiClient.php]: make sense Done! :)

**Comment by elminson** [comment]
fixing the failing standards 
@scheb my PHP standard is set to use tabs :) fixed

**Comment by scheb** [comment]
Merged and tagged as v4.9.0

**Pull request merged**